crates/astria-composer/src/executor/mod.rs

```diff
@@ -212,25 +214,26 @@ impl Executor {
     ///
     /// # Errors
     /// An error is returned if connecting to the sequencer fails.
-    // Allow: Refactored in <insert GH here>. Current logging events will show as errors in otel,
-    // refactor addresses this as well.
-    #[allow(clippy::too_many_lines)]
     pub(super) async fn run_until_stopped(mut self) -> eyre::Result<()> {
         select!(
             biased;
             () = self.shutdown_token.cancelled() => {
-                info!("received shutdown signal while running initialization routines; exiting");
-                return Ok(());
+                log_exit(&Ok("received shutdown signal while running initialization routines; exiting"));
+                return Ok(())
             }
 
             res = self.pre_run_checks() => {
+                log_exit(&res);
                 res.wrap_err("required pre-run checks failed")?;
             }
         );
         let mut submission_fut: Fuse<Instrumented<SubmitFut>> = Fuse::terminated();
         let mut nonce = get_latest_nonce(self.sequencer_client.clone(), self.address, self.metrics)
```

Comment:
If we refactor this anway, please combine this `get_latest_nonce` with the `self.pre_run_checks()` above by creating something like a `Executor::init` method.

Right now `get_latest_nonce` ignores the shutdown token above and this PR looks like a good place to fix that.

Reply:
Refactored such that `ensure_chain_id_is_correct()` and `get_latest_nonce()` are in an `Executor::init()` function, which will call `report_exit_reason()` on an Err result.